starter-code/scripts/articleView.js

```diff
@@ -37,36 +38,53 @@ articleView.handleAuthorFilter = function() {
     //         defining. "$(this)" is using jQuery to select that element, so we can chain jQuery methods
     //         onto it.
     if ($(this).val()) {
-      // TODO: If the select box was changed to an option that has a value, we need to hide all the articles,
+      // Done: If the select box was changed to an option that has a value, we need to hide all the articles,
       //       and then show just the ones that match for the author that was selected.
       //       Use an "attribute selector" to find those articles, and fade them in for the reader.
+      $('article').filter((ind, art) => $(art).attr('data-author') != $(this).val()).each(
+        (ind, art) =>$(art).hide());
 
+      var selectedItem = $('article').filter((ind, art) => $(art).attr('data-author') === $(this).val())[0];
+      $(selectedItem).show();
     } else {
-      // TODO: If the select box was changed to an option that is blank, we should
+      // Done: If the select box was changed to an option that is blank, we should
       //       show all the articles, except the one article we are using as a template.
-
+      $('article').each((ind, art) => $(art).show());
     }
     $('#category-filter').val('');
   });
 };
 
 articleView.handleCategoryFilter = function() {
-  // TODO: Just like we do for #author-filter above, we should handle change events on the #category-filter element.
-  //       When an option with a value is selected, hide all the articles, then reveal the matches.
-  //       When the blank (default) option is selected, show all the articles, except for the template.
-  //       Be sure to reset the #author-filter while you are at it!
+  // Done!!
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').filter((ind, art) => $(art).attr('data-category') != $(this).val()).each(
+        (ind, art) =>$(art).hide());
 
+      var selectedItem = $('article').filter((ind, art) => $(art).attr('data-category') === $(this).val())[0];
+      $(selectedItem).show();
```

Comment:
This only shows the first article that has the given category, same with the author filter above, but it is less obvious since there is only one article by each author. You need to be able to show all the articles that match the given category.